hack/compare-k8s-feature-gates.sh

```diff
@@ -0,0 +1,61 @@
+#!/bin/bash
+#
+# Copyright (c) 2021 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -e
+
+usage() {
+  echo "Usage:"
+  echo "> compare-k8s-feature-gates.sh [ -h | <old version> <new version> ]"
+
+  exit 0
+}
+
+if [ "$1" == "-h" ] || [ "$#" -ne 2 ]; then
+  usage
+fi
+
+versions=("$1" "$2")
+files=(
+  "pkg/features/kube_features.go"
+  "staging/src/k8s.io/apiserver/pkg/features/kube_features.go"
+  "staging/src/k8s.io/apiextensions-apiserver/pkg/features/kube_features.go"
+  "staging/src/k8s.io/controller-manager/pkg/features/kube_features.go"
```

Comment:
We could add some special handling for feature gates present in `staging/src/k8s.io/controller-manager/pkg/features/kube_features.go` but not in `pkg/features/kube_features.go`, as they will only be available for kcm.
E.g. `ControllerManagerLeaderMigration` is only available in kcm, but not in k-api (I think, it's the only one in the 1.21, so don't know if it's worth the effort). Setting it on k-api will make it fail:
```
$ docker run k8s.gcr.io/kube-apiserver:v1.21.1 /usr/local/bin/kube-apiserver --feature-gates ControllerManagerLeaderMigration=true
[...]
Error: invalid argument "ControllerManagerLeaderMigration=true" for "--feature-gates" flag: unrecognized feature gate: ControllerManagerLeaderMigration
2021/06/04 09:11:09 running command: exit status 1
```

Reply:
Also here, we discussed this with @rfranzke and explicitly decided against it. The enhanced "solidity" comes at additional complexity and maintenance cost. Besides min and max versions, it would require maintaining a component for which the feature gate is valid, to be able to perform the check. Then the comparison script would need to extract this information, and the maintainer would have to maintain it for every new version. Having in mind that the chance of hitting upon this problem is quite low (here even lower than for "locked to default"), we decided it's not worth it.

Reply:
Ok, I understand. Didn't know, you already discussed this.
Fine for me then.